lib/straight-server/gateway.rb

```diff
@@ -61,7 +61,13 @@ def initialize_blockchain_adapters
       end
       @blockchain_adapters      = all_instances[:main]
       @test_blockchain_adapters = all_instances[:test]
-      raise NoBlockchainAdapters if all_instances.values.all?(&:empty?)
+      if all_instances.values.all?(&:empty?)
+        raise NoBlockchainAdapters
+      elsif @blockchain_adapters.empty?
+        StraightServer.logger.warn "No mainnet blockchain adapters are configured"
+      elsif @test_blockchain_adapters.empty?
+        StraightServer.logger.warn "No testnet blockchain adapters are configured"
+      end
```

Comment:
I think that there are a lot of code for this method. Maybe we should move it to other functions or validators?